dateutil/parser/_parser.py

```diff
@@ -60,14 +60,8 @@ class _timelex(object):
     _split_decimal = re.compile("([.,])")
 
     def __init__(self, instream):
-        if six.PY2:
-            # In Python 2, we can't duck type properly because unicode has
-            # a 'decode' function, and we'd be double-decoding
-            if isinstance(instream, (bytes, bytearray)):
-                instream = instream.decode()
-        else:
-            if getattr(instream, 'decode', None) is not None:
-                instream = instream.decode()
+        if isinstance(instream, (bytes, bytearray)) or six.PY3 and isinstance(instream, memoryview):
```

Comment:
I think that what @vstinner meant with supporting memoryview was not this. This will fail when a memoryview is passed, as it does not have a `decode` method AFAIK.

If you want to add support for memoryview, please add tests for it. Alternatively, feel free to just remove `or six.PY3 and isinstance(instream, memoryview)` from this PR.